### PR TITLE
feat(core): address oa todos

### PR DESCRIPTION
**Pull request by Reinis-FRP**
Signed-off-by: Reinis Martinsons <reinis@umaproject.org>

**Summary**

Reviews and addresses remaining Optimistic Asserter TODOs.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol [hunk 1]
```diff
@@ -91,8 +91,6 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable, M
         uint64 liveness,
         bytes32 identifier
     ) public nonReentrant returns (bytes32) {
-        // TODO: think about placing either msg.sender or block.timestamp into the claim ID to block an advasery
-        // creating a claim that collides with a known assertion that will be created into the future.
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol:95]: Addressed in #4290

File: packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol [hunk 2]
```diff
@@ -185,7 +183,6 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable, M
         _oracleRequestPrice(assertionId, assertion.identifier, assertion.assertionTime);
 
         // Send dispute callback
-        // TODO: consider mergeing the isDisputeAlloowed into toe SSM callback (revert within callback to block).
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol:188]: Not implementing yet as most of applications would not use disputer filters. If we were to do this then a dedicated dispute callback allowing to pass `msg.sender` as `disputeCaller` and `validateDisputers` property should be added to Escalation Manager interface.

File: packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol [hunk 3]
```diff
@@ -4,9 +4,6 @@ pragma solidity 0.8.16;
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 interface OptimisticAsserterInterface {
-    // TODO: SecurityManager!!!. rename EscalationManagerManager->EscalationManager
-    // TODO: Em->EscalationManager...check this is not too long.
-    // Options: AssertionManager, EscalationManager, SecurityManager, SovereignManager.
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol:9]: Addressed in #4280

File: packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol [hunk 4]
```diff
@@ -27,7 +23,7 @@ interface OptimisticAsserterInterface {
         uint256 bond;
         bool settled; // True if the request is settled.
         bool settlementResolution;
-        uint64 assertionTime; // Time of the assertion. TODO uint64 could be enough.
+        uint64 assertionTime; // Time of the assertion.
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-asserter/interfaces/OptimisticAsserterInterface.sol:26]: Addressed in #4285

**Review by md0x — APPROVED** [review]
👌

**Pull request merged**